java/src/main/java/com/logicalclocks/hsfs/TrainingDataset.java

```diff
@@ -319,7 +320,20 @@ public void show(int numRows) throws FeatureStoreException, IOException {
    */
   public Statistics computeStatistics() throws FeatureStoreException, IOException {
     if (statisticsConfig.getEnabled()) {
-      return statisticsEngine.computeStatistics(this, read());
+      if (this.splits != null && !this.splits.isEmpty()) {
+        List<SplitStatistics> splitStatistics = new ArrayList<>();
+        for (Split split : this.splits) {
+          splitStatistics.add(new SplitStatistics(split.getName(),
+              statisticsEngine.computeStatistics(read(split.getName()),
+                  this.getStatisticsConfig().getColumns(),
+                  this.getStatisticsConfig().getHistograms(),
+                  this.getStatisticsConfig().getCorrelations(),
+                  null).getContent()));
+        }
+        return statisticsEngine.registerSplitStatistics(this, splitStatistics);
```

Comment:
This violates our convention of having all the computation and logic on the engines classes. I'd prefer if we don't break this convention as it worked quite well so far. I'd move the split iteration in the engine.

Reply:
I just leave it here as a reminder -  when https://github.com/logicalclocks/feature-store-api/pull/407 gets merged, we have to pass the `exactUniqueness` parameter here as well.